From: SeongJae Park <sj@kernel.org>
Subject: mm/damon/core: hide damon_add_region()
Date: Fri, 22 May 2026 08:40:16 -0700

damon_add_region() is being used by only DAMON core, but exposed to DAMON
API callers.  Exposing something that is not really being used by others
will only increase the maintenance cost.  Hide it.

Link: https://lore.kernel.org/20260522154026.80546-6-sj@kernel.org
Signed-off-by: SeongJae Park <sj@kernel.org>
Cc: Brendan Higgins <brendan.higgins@linux.dev>
Cc: Shuah Khan <shuah@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/damon.h |    1 -
 mm/damon/core.c       |    2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

--- a/include/linux/damon.h~mm-damon-core-hide-damon_add_region
+++ a/include/linux/damon.h
@@ -1024,7 +1024,6 @@ static inline void damon_insert_region(s
 	t->nr_regions++;
 }
 
-void damon_add_region(struct damon_region *r, struct damon_target *t);
 void damon_destroy_region(struct damon_region *r, struct damon_target *t);
 int damon_set_regions(struct damon_target *t, struct damon_addr_range *ranges,
 		unsigned int nr_ranges, unsigned long min_region_sz);
--- a/mm/damon/core.c~mm-damon-core-hide-damon_add_region
+++ a/mm/damon/core.c
@@ -250,7 +250,7 @@ struct damon_region *damon_new_region(un
 	return region;
 }
 
-void damon_add_region(struct damon_region *r, struct damon_target *t)
+static void damon_add_region(struct damon_region *r, struct damon_target *t)
 {
 	list_add_tail(&r->list, &t->regions_list);
 	t->nr_regions++;
_
